models/cnn/AlexNet/test_AlexNet.py

```diff
@@ -1,21 +1,37 @@
+import pytest
 from AlexNet import AlexNet
 import tensorflow as tf
 
 import logging
 logging.basicConfig(format='%(asctime)s [%(levelname)s] %(message)s', level=logging.DEBUG)
 
 
-def test_forward_pass(model):
+@pytest.mark.dependency()
+@pytest.mark.parametrize("input_shape", [
+    (128, 128, 1), (128, 128, 3), (256, 256, 3),
+])
+
+def test_forward_pass(input_shape):
     logging.info(f"[RUNNING] Test case 'forward_pass' started.")
+    model = AlexNet(num_classes=10)
+    model = model.build(input_shape=input_shape)
     input = tf.zeros(shape=(1,)+model.layers[0].input_shape[0][1:])
+    logging.debug(f"{input.shape=}")
     output = model(input)
     assert len(output.shape) == 2, f"{output.shape=}"
     assert output.shape[0] == input.shape[0], f"{output.shape=}"
     logging.info("[PASSED] Test case 'forward_pass' passed.")
 
-
-def test_overfit(model):
+@pytest.mark.dependency(depends=['test_forward_pass'])
+@pytest.mark.slow
+@pytest.mark.parametrize("input_shape", [
+    (128, 128, 3),
+])
+def test_overfit(input_shape):
     logging.info(f"[RUNNING] Test case 'overfit' started.")
+    model = AlexNet(num_classes=10)
+    model = model.build(input_shape=input_shape)
+    loss = tf.keras.losses.SparseCategoricalCrossentropy()
     model.compile(
         loss=tf.keras.losses.SparseCategoricalCrossentropy(),
```

Comment:
you don't need to repeat `tf.keras.losses.SparseCategoricalCrossentropy()`. just pass the variable `loss` above.